lib/1.4/dml-builtins.dml

```diff
@@ -532,15 +532,24 @@ template device {
 
 Group objects are generic container objects, used to group other objects.
 They can appear anywhere in the object hierarchy, but some object types
-(currently `implement`, `interface`, `bank` and `port`) may not have a group
+(currently `implement` and `interface`) may not have a group
 as parent.
 
 The `group` template contains no particular methods or parameters other
 than what is inherited from the [`object`](#object) template.
+
+In a group hierarchy directly below a `port` or `bank` object, none of the
+group object may be named `device` or `port`; this is to avoid name clashes.
```

Comment:
This is mixed-up compared to the check: the check forbids a group from being named "port" or "bank" if the group's a part of a `port` or `device`.

Reply:
thanks, fixed